pkg/virt-api/rest/spice.go

```diff
@@ -72,28 +67,13 @@ func spiceFromVM(vm *v1.VM, coreCli *kubernetes.Clientset) (*v1.Spice, error) {
 	}
 
 	// TODO allow specifying the spice device. For now select the first one.
-	for _, d := range vm.Spec.Domain.Devices.Graphics {
-		if strings.ToLower(d.Type) == "spice" {
-			port := d.Port
-			podList, err := coreCli.CoreV1().Pods(api.NamespaceDefault).List(unfinishedVMPodSelector(vm))
-			if err != nil {
-				return nil, middleware.NewInternalServerError(err)
-			}
-
-			// The pod could just have failed now
-			if len(podList.Items) == 0 {
-				// TODO is that the right return code?
-				return nil, middleware.NewResourceNotFoundError("VM is not running")
-			}
-
-			pod := podList.Items[0]
-			ip := pod.Status.PodIP
-
+	for _, d := range vm.Status.Graphics {
+		if d.Type == "spice" {
 			spice := v1.NewSpice(vm.GetObjectMeta().GetName())
 			spice.Info = v1.SpiceInfo{
 				Type: "spice",
-				Host: ip,
-				Port: port,
+				Host: d.Host,
```

Comment:
To make seamless spice work during migrations, you also have to change things at https://github.com/kubevirt/kubevirt/blob/master/pkg/virt-controller/services/template.go#L133. Otherwise during the migration, the spice connection would be forwarded to the pod ip, which is no longer the case in this PR.